### DISCUSSION TITLE
Refresh streaming connections on signIn / signOut

### DISCUSSION THREAD

**Discussion by Disregard-Therest** [serverpod/serverpod#2933] [Q&A]
Hi,
for example chat_endpoint sets UserObject when the stream is Opened

` Future<void> streamOpened(StreamingSession session) async {
    var userId = (await session.authenticated)?.userId;

    if (userId != null) {
      setUserObject(
          session,
          _ChatSessionInfo(
            userInfo: await Users.findUserByUserId(session, userId),
          ));
    } else {
      setUserObject(session, _ChatSessionInfo());
    }
} `

However, when the user sign in or signs out - this method is not called again. So chats actually don't work correctly.

Like a workaround our team needs to close connection when the user signs out and open again when sign in is finished.
But I think that your team should have some better decision on this.

**Comment by Disregard-Therest** [comment]
Actually something wrong with sockets

This thing is called, but nothing happens
![image](https://github.com/user-attachments/assets/d72f0f47-d009-4e68-911a-78334d146c02)


This code on the server isn't called after signIn/signOut unless I close streaming connection
![image](https://github.com/user-attachments/assets/b8d086bb-d987-4a4c-98b2-25ab0700f35c)

**Reply by SandPod** [reply]
Hey @Disregard-Therest!

The code in your second image is only used to establish the streaming connection.

The first image caputers where we send a control command to authenticate the stream. This is processed a bit further down in the second image, more specifically here: https://github.com/serverpod/serverpod/blob/939cb3899311a9eb79f1a567b2d2875e6b6e5b60/packages/serverpod/lib/src/server/websocket_request_handlers/endpoint_websocket_request_handler.dart#L60-L63

**Reply by Disregard-Therest** [reply]
Ok, I've finally got it. Clear now, but the problem is that these lines don't affect anything, because real impact is in the _callStreamOpened - every endpoint sets it's objects and configures channels.

So actually the code from lines 30 - 40, should be somehow executed again if the auth changes.

**Reply by SandPod** [reply]
Hmm, you might be right there, you need to be authenticated before the streaming connection to the server is established. So I would suggest doing a reconnect if the authentication information in the client changes.

We are looking at migrating the chat module to the new streaming methods. In that migration we will fix these sort of issues.
